summary-v4-object.go

```diff
@@ -0,0 +1,166 @@
+//
+// Copyright (c) 2015-2025 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+package madmin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// ObjectSummaryOptions ...
+type ObjectSummaryOptions struct {
+	Bucket, Object string
+}
+
+// ObjectSummary calls minio to search for all files and parts
+// related to the given object, across all disks.
+func (adm *AdminClient) ObjectSummary(ctx context.Context, objOpts ObjectSummaryOptions) (objectSummaryResponse *ObjectSummaryRespose, err error) {
+	form := make(url.Values)
+	if objOpts.Bucket != "" {
+		form.Add("bucket", objOpts.Bucket)
+	}
+	if objOpts.Object != "" {
+		form.Add("object", objOpts.Object)
+	}
+
+	resp, err := adm.executeMethod(ctx,
+		http.MethodGet,
+		requestData{
+			relPath:     fmt.Sprintf(adminAPIPrefix + "/object-info"),
+			queryValues: form,
+		})
+	defer closeResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, httpRespToErrorResponse(resp)
+	}
+
+	objectSummaryResponse = new(ObjectSummaryRespose)
+	dec := json.NewDecoder(resp.Body)
```

Comment:
why do we care about JSON? can we get this in msgpack instead?

Reply:
worried if this becomes too big, because you are also using 

```
	Files map[string]*ObjectSummaryInfo
```

which can be un-ending amount of files if someone tries to get this information across namespace.

Reply:
The idea was to enable multiple object support later, it's essentially the same kind of "danger" as inspect object. But thinking about it, users will be using this much so I should probably limit this to a single object for now and make sure it won't just walk infinite depth

Reply:
I'll also move this to msgpack